packages/xo-web/src/xo-app/new/network/index.js

```diff
@@ -0,0 +1,244 @@
+import decorate from 'apply-decorators'
+import Page from '../../page'
+import PropTypes from 'prop-types'
+import styles from './index.css'
+import React, { Component } from 'react'
+import Wizard, { Section } from 'wizard'
+import { connectStore } from 'utils'
+import { createBondedNetwork, createNetwork, getBondModes } from 'xo'
+import { createGetObject, getIsPoolAdmin } from 'selectors'
+import { injectIntl } from 'react-intl'
+import { injectState, provideState } from 'reaclette'
+import { linkState } from 'reaclette-utils'
+import { map } from 'lodash'
+import { Select, Toggle } from 'form'
+import { SelectPif, SelectPool } from 'select-objects'
+
+const EMPTY = {
+  bonded: false,
+  bondMode: undefined,
+  description: '',
+  mtu: '',
+  name: '',
+  pif: undefined,
+  pifs: [],
+  vlan: '',
+}
+
+const NewNetwork = decorate([
+  connectStore(() => ({
+    isPoolAdmin: getIsPoolAdmin,
+    pool: createGetObject((_, props) => props.location.query.pool),
+  })),
+  injectIntl,
+  provideState({
+    initialState: () => ({ ...EMPTY, bondModes: undefined }),
+    effects: {
+      initialize: async () => ({ bondModes: await getBondModes() }),
+      linkState,
+      onChangeMode: (_, bondMode) => ({ bondMode }),
+      onChangePif: (_, value) => ({ bonded }) =>
+        bonded ? { pifs: value } : { pif: value },
+      reset: () => EMPTY,
+      toggleBonded: () => ({ bonded }) => ({
+        ...EMPTY,
+        bonded: !bonded,
+      }),
+    },
+    computed: {
+      modeOptions: ({ bondModes }) =>
+        bondModes !== undefined
+          ? bondModes.map(mode => ({
+              label: mode,
+              value: mode,
+            }))
+          : [],
+      pifPredicate: (_, { pool }) => pif =>
+        pif.vlan === -1 && pif.$host === (pool && pool.master),
+    },
+  }),
+  injectState,
+  class extends Component {
+    static contextTypes = {
+      router: PropTypes.object,
+    }
+
+    _create = () => {
+      const { pool, state } = this.props
+      const {
+        bonded,
+        bondMode,
+        description,
+        mtu,
+        name,
+        pif,
+        pifs,
+        vlan,
+      } = state
+      return bonded
+        ? createBondedNetwork({
+            bondMode: bondMode.value,
+            description,
+            mtu,
+            name,
+            pifs: map(pifs, 'id'),
+            pool: pool.id,
+            vlan,
+          })
+        : createNetwork({
+            description,
+            mtu,
+            name,
+            pif: pif.id,
+            pool: pool.id,
+            vlan,
+          })
+    }
+
+    _selectPool = pool => {
+      const {
+        effects,
+        location: { pathname },
+      } = this.props
+      effects.reset()
+      this.context.router.push({
+        pathname,
+        query: pool !== null && { pool: pool.id },
+      })
+    }
+
+    _redirectOnSuccess = network =>
+      network !== undefined ? `pools/${this.props.pool.id}/network` : undefined
```

Comment:
If the creation of network was failed.

Reply:
Doesn't `redirectOnSuccess` redirect only on success?

Reply:
the `redirectOnSuccess` is defined without `network !== undefined`. [we haven't condition for result](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/action-button.js#L97) .

Reply:
I'm not sure to understand what you mean. If the [actions fails](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/action-button.js#L92), it will not redirect.